### PR TITLE
Reports: Comma-formatted numbers

### PR DESCRIPTION
**Pull request by claudiovallejo**
**Story card:** [ch1824](https://app.clubhouse.io/simpledotorg/story/1824/comma-separated-numbers)

## Because

We should display numbers in a consistent format across the Dashboard. This PR ensures numbers are:
1. Displayed as `-` if `0`
2. Comma-formats numbers (i.e. `1234` → `1,234`)

## This addresses

Displays comma-formatted numbers in the following Dashboard sections

1. Reports "Details"
2. Reports "Cohort reports"

### CODE REVIEW

File: app/views/reports/regions/details.html.erb [hunk 1]
```diff
@@ -59,16 +59,16 @@
                 Total
               </td>
               <td class="row-total ta-center">
-                <%= dash_if_zero(@dashboard_analytics.sum { |_, row| row[:total_registered_patients] || 0 }) %>
```
Comment (harimohanraj89) [inline app/views/reports/regions/details.html.erb:62]: We're no longer using dashes if the number is zero? Is that intentional?

**Review by harimohanraj89 — CHANGES_REQUESTED** [review]
One question from me.

File: app/views/reports/regions/details.html.erb (hunk 1 above)
Comment (claudiovallejo) [reply app/views/reports/regions/details.html.erb:62]: I did this intentionally, as it's the only table where we do this. I will get Daniel's approval first though.

Comment (harimohanraj89) [reply app/views/reports/regions/details.html.erb:62]: Looks like we want to stick with dashes according to Slack.

I did a quick check and it's safe to pass dashes to the `number_with_delimiter` methods, so we can update these lines to:
```erb
<%= number_with_delimiter(dash_if_zero(@dashboard_analytics.blah_blah)) %>
```

Comment (claudiovallejo) [reply app/views/reports/regions/details.html.erb:62]: Great, thank you @harimohanraj89 !

**Review by harimohanraj89 — APPROVED** [review]

**Pull request merged**